File: src/main/java/imaxct/service/impl/CourseServiceImpl.java
package imaxct.service.impl;

import imaxct.domain.Course;
import imaxct.service.ICourseService;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Created by maxct on 2017/4/9.
 */
@Service
@Transactional
public class CourseServiceImpl extends BaseService implements ICourseService {

    public boolean addCourse(Course course) {
        return courseDao.addCourse(course);
    }

    public boolean deleteCourse(Course course) {
        return courseDao.deleteCourse(course);
    }

    public List<Course> getAllCourses(int id) {
        if (id == -1)
            return this.courseDao.getAllCourses();
        return courseDao.getCourseFromId(id);
    }

}

File: src/main/java/imaxct/service/impl/CourseServiceImpl.kt
package imaxct.service.impl

import imaxct.bean.Msg
import imaxct.domain.Course
import imaxct.service.ICourseService
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
 * Created by maxct on 2017/4/9.
 */
@Service
@Transactional
open class CourseServiceImpl : BaseService(), ICourseService {
    override fun updateCourse(course: Course): Boolean = this.courseDao!!.updateCourse(course)

    override fun getCourseById(id: Int): Course? = this.courseDao!!.getCourseById(id)

    override fun addCourse(course: Course): Msg<*> {
        if (courseDao!!.addCourse(course))
            return Msg(0, "ok", null)
        return Msg(-1, "添加失败.", null)
    }

    override fun deleteCourse(course: Course): Msg<*> {
        if (courseDao!!.deleteCourse(course))
            return Msg(0, "ok", null)
        return Msg(-1, "删除失败.", null)
    }

    override fun getAllCourses(id: Int): Msg<List<Course>> {
        if (id == -1)
            return Msg(0, "ok", this.courseDao!!.getAllCourses())
        return Msg(0, "ok", this.courseDao!!.getCourseFromId(id))
    }

}
